File: source/python/0020/0020.py
# n! means n  (n  1)  ...  3x2x1
#
# For example, 10! = 10x9x...x3x2x1 = 3628800,
# and the sum of the digits in the number 10! is
# 3 + 6 + 2 + 8 + 8 + 0 + 0 = 27.
#
# Find the sum of the digits in the number 100!
import operator
print reduce(lambda x,y:int(x)+int(y),str(reduce(operator.mul,xrange(2,101))))

File: source/haskell/0020/0020.hs
-- n! means n x (n - 1) x ...x 3 x 2 x 1
--
-- For example, 10! = 10 x 9 x ... x 3 x 2 x 1 = 3628800,
-- and the sum of the digits in the number 10! is
-- 3 + 6 + 2 + 8 + 8 + 0 + 0 = 27.
--
-- Find the sum of the digits in the number 100!
import Data.Char
main = print $ foldr (+) 0 $ map digitToInt $ show $ foldr (*) 1 [2..100]  
